migration-workflow.md

```diff
@@ -0,0 +1,72 @@
+# Migrating Fiftyone Teams Auth
+This document will help operators migrate an existing Fiftyone Teams
+deployment from `legacy` mode to `internal` mode. This will remove the
+dependency on Auth0 as a user management system.
+
+## Terms
+Fiftyone Teams now has two modes: `legacy` and `internal`.
+
+`legacy` mode is fully dependent on an Auth0 connection to handle
+authentication and user management.
+
+By contrast, `internal` mode functions by moving user management to within
+the system. After an initial migration of users, operators of Fiftyone Teams
+may continue to use Auth0 as an identity provider (IdP) to facilitate
+authentication while keeping all other aspects of user management internal.
+
+Additionally, operators may choose to change the target IdP to be something
```

Comment:
Using Auth0 is just a stopgap so that users can start accessing the system immediately. Before CAS and internal mode, no IdP was required after the initial deployment phase, so there wasn't one configured and operators didn't need to think about it.

The migration script is going to make sure that when it is done, _at least_ Auth0 is provided to the CAS, since it's established already and the users can be auth'd that way. So even if they have another IdP they prefer, they don't need to worry about configuring it with FoT until they want to, because we can use the existing Auth0 in the meantime so there's no disruption of service -- even  if most of the time that's just in turn passing through to whatever was used during their deployment.